### PR TITLE
Updated Decimal_to_Binary.cpp

### LINKED ISSUE #964: Number Conversion (Decimal To Binary)
can i start working on it ?

### PR DESCRIPTION
**Pull request by Manas2520**
Fixes #964 

#### Checklist:
- [x] 4 space indentation.
- [x] Coding conventions are followed.
- [x] Input is taken dynamically.
- [x] Sample Input / Output is added at the end of file.
- [x] Logic Documentation (Comments).
- [x] File names are correct.

Make sure these boxes are checked before your pull request (PR) is ready to be reviewed and merged. Thanks!

#### Changes proposed in this pull request:
Added Decimal_to_Binary.cpp

#### Languages Used:
C++

#### Files Added:
Decimal_to_Binary.cpp

> We're happy to help you get this ready -- don't be afraid to ask for help.

Thanks!

### CODE REVIEW

**Comment by Manas2520** [comment]
Please review it @singh-shreya6

File: Number_Conversion/C++/Decimal_to_Binary.cpp [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+#include <math.h>
+using namespace std;
+
+// Function to convert the number from decimal to binary form
+void printBinary(int n){
+
+  int size = log2(n) + 1;  // command to find out number of bits in the given number
+                          // which decides the size of the array
+  int b[size], j = 0;    // Array to store the binary number
+
```
Comment (singh-shreya6) [inline Number_Conversion/C++/Decimal_to_Binary.cpp:12]: Please ensure 4 space indentation here. It is only 2 spaces.

File: Number_Conversion/C++/Decimal_to_Binary.cpp [hunk 2]
```diff
@@ -0,0 +1,37 @@
+// C++ program to convert numbers from decimal to binary form
+#include <iostream>
+#include <math.h>
+using namespace std;
+
+// Function to convert the number from decimal to binary form
+void printBinary(int n){
+
+  int size = log2(n) + 1;  // command to find out number of bits in the given number
```
Comment (singh-shreya6) [inline Number_Conversion/C++/Decimal_to_Binary.cpp:9]: Please put the side comment before the line for which it is a comment e.g
// Array to store the binary number
int b[size], j = 0;

File: Number_Conversion/C++/Decimal_to_Binary.cpp [hunk 3]
```diff
@@ -0,0 +1,37 @@
+// C++ program to convert numbers from decimal to binary form
+#include <iostream>
```
Comment (singh-shreya6) [inline Number_Conversion/C++/Decimal_to_Binary.cpp:2]: please leave one blank line before this.

**Comment by singh-shreya6** [comment]
Also in the PR template, don't add fixes# to the last PR. Add Fixes#964 only, the original issue.

**Comment by Manas2520** [comment]
Ok ma'am. Thank you for bearing with me

**Comment by singh-shreya6** [comment]
Please squash your commits @Manas2520

**Comment by Manas2520** [comment]
Ma'am I am facing a problem.When you asked me to make the changes,I went to the edit file option in the pull request and made the changes and commited those within the PR. However, while rebasing it is not showing that commit.What should I do?

**Comment by singh-shreya6** [comment]
As shared by one of the participants, please follow this procedure to squash commits:
Suppose the name of my working branch is my_branch

I will switch to my branch
git checkout my_branch

After that, I will do all my work in that branch. I will modify files, work on them and finally push all the commits and changes.

Now I need to squash the commits [right now I am in my_branch]
git rebase -i upstream/master

A screen will open up showing the list of all commits with different labels along with it.
Keep the first commit as pick
& change all other commits to squash

This will squash (merge) the respective commit onto previous one
After changing all labels to squash except the first one, save & exit.

Next, a screen will appear showing all commit messages, If you want, you can change the commit messages or leave them as it is. Save and exit.

This will rebase all the commits in case there are no merge conflicts.

Last step is to force push the commits so that commit count becomes one.
git push --force origin my_branch

Done.

**Comment by Manas2520** [comment]
Ma'am,I did it somehow,but it left me feeling confused.Please see if its done correctly.

**Comment by Manas2520** [comment]
Thank you ma'am.Now I understand how to squash commits.The only problem is that in the process of experimentation I squashed a few commits that were not mine.

**Comment by singh-shreya6** [comment]
Hey your commit count is one but there are other files now in your files changed. You will have to remove those from your branch. First take the latest version of code by using git fetch command mentioned in codebase. Then only add your change to your local repository and commit. After that follow these steps. If it is getting very messy, then open a new PR and close this one as you have already learnt how to squash commits.

**Comment by Manas2520** [comment]
Thank you ma'am,for clearing my confusion. I will open a new PR. Please review it.

**Pull request closed without merging**